### ISSUE TITLE
Local test failures from 'cryst_prop_pack.py'

### ISSUE THREAD

**Issue by aladshaw3** [watertap-org/watertap#341]
On fresh reinstall of watertap (using `requirements-dev.txt`) there are 2 test failures from the `cryst_prop_pack.py`

```
FAILED tests\test_cryst_prop_pack.py::TestNaClProperty_idaes::test_state_block_mole_frac_phase_comp - ValueError: mole_frac_phase_comp is not indexed by phase and compon...

FAILED tests\test_cryst_prop_pack.py::TestNaClProperty_idaes::test_get_material_flow_terms - KeyError: 'get_material_flow_terms method is not indexed by phase and compon...
```

These fail on both Linux and Windows for me locally.

**Comment by aladshaw3** [comment]
[test_failures.txt](https://github.com/watertap-org/watertap/files/8100592/test_failures.txt)

**Comment by lbianchi-lbl** [comment]
@aladshaw3, thanks for opening the issue. Can you post the output of these commands from the same environment?

- `pip list`
- `idaes environment-info`

**Comment by aladshaw3** [comment]
(watertap-dev) PS C:\Users\1pi> pip list
Package                            Version     Location
---------------------------------- ----------- ------------------------------
alabaster                          0.7.12
argon2-cffi                        21.1.0
atomicwrites                       1.4.0
attrs                              21.2.0
Babel                              2.9.1
backcall                           0.2.0
backports.shutil-get-terminal-size 1.0.0
black                              21.9b0
bleach                             4.1.0
bunch                              1.0.1
certifi                            2021.10.8
cffi                               1.15.0
charset-normalizer                 2.0.7
click                              7.1.2
colorama                           0.4.4
coverage                           6.0.2
cycler                             0.10.0
dataclasses-json                   0.5.6
debugpy                            1.5.0
decorator                          5.1.0
defusedxml                         0.7.1
docutils                           0.17.1
entrypoints                        0.3
fastjsonschema                     2.15.1
Flask                              2.0.2
Flask-Cors                         3.0.10
htmlmin                            0.1.12
idaes-pse                          1.13.0.dev0
idna                               3.3
imagesize                          1.2.0
iniconfig                          1.1.1
ipykernel                          6.4.1
ipython                            7.28.0
ipython-genutils                   0.2.0
ipywidgets                         7.6.5
itsdangerous                       2.0.1
jedi                               0.18.0
Jinja2                             3.0.2
json-schema-for-humans             0.34.4
jsonschema                         4.1.0
jupyter                            1.0.0
jupyter-client                     7.0.6
jupyter-console                    6.4.0
jupyter-core                       4.8.1
jupyterlab-pygments                0.1.2
jupyterlab-widgets                 1.0.2
kiwisolver                         1.3.2
linkify-it-py                      1.0.2
lxml                               4.6.3
markdown-it-py                     1.1.0
markdown2                          2.4.1
MarkupSafe                         2.0.1
marshmallow                        3.13.0
marshmallow-enum                   1.5.1
matplotlib                         3.4.3
matplotlib-inline                  0.1.3
mdit-py-plugins                    0.2.8
mistune                            0.8.4
mongomock                          3.23.0
mpmath                             1.2.1
mypy-extensions                    0.4.3
myst-parser                        0.15.2
nbclient                           0.5.4
nbconvert                          6.2.0
nbformat                           5.1.3
nbsphinx                           0.8.8
nest-asyncio                       1.5.1
networkx                           2.6.3
notebook                           6.4.4
numpy                              1.21.2
omlt                               0.3.0
packaging                          21.0
pandas                             1.3.3
pandocfilters                      1.5.0
parso                              0.8.2
pathspec                           0.9.0
pickleshare                        0.7.5
Pillow                             8.4.0
Pint                               0.17
pip                                21.1.3
platformdirs                       2.4.0
pluggy                             1.0.0
ply                                3.11
prometheus-client                  0.11.0
prompt-toolkit                     3.0.20
psutil                             5.8.0
py                                 1.10.0
pycparser                          2.20
Pygments                           2.10.0
pymongo                            3.12.0
Pyomo                              6.2
pyparsing                          2.4.7
pyrsistent                         0.18.0
pytest                             6.2.5
pytest-cov                         3.0.0
python-dateutil                    2.8.2
python-slugify                     5.0.2
pytz                               2021.3
pywin32                            225
pywinpty                           1.1.4
PyYAML                             6.0
pyzmq                              22.3.0
qtconsole                          5.1.1
QtPy                               1.11.2
rbfopt                             4.2.2
regex                              2021.10.8
requests                           2.26.0
scipy                              1.7.1
Send2Trash                         1.8.0
sentinels                          1.0.0
setuptools                         58.2.0
six                                1.16.0
snowballstemmer                    2.1.0
Sphinx                             4.2.0
sphinx-rtd-theme                   1.0.0
sphinxcontrib-applehelp            1.0.2
sphinxcontrib-devhelp              1.0.2
sphinxcontrib-htmlhelp             2.0.0
sphinxcontrib-jsmath               1.0.1
sphinxcontrib-qthelp               1.0.3
sphinxcontrib-serializinghtml      1.1.5
sympy                              1.9
terminado                          0.12.1
testpath                           0.5.0
text-unidecode                     1.3
tinydb                             4.5.2
toml                               0.10.2
tomli                              1.2.1
tornado                            6.1
traitlets                          5.1.0
typing-extensions                  3.10.0.2
typing-inspect                     0.7.1
uc-micro-py                        1.0.1
urllib3                            1.26.7
watertap                           0.4.0.dev0  c:\users\1pi\projects\watertap
wcwidth                            0.2.5
webencodings                       0.5.1
Werkzeug                           2.0.2
wheel                              0.37.0
widgetsnbextension                 3.5.1

**Comment by aladshaw3** [comment]
The `idaes environment-info` command returns an error.

(watertap-dev) PS C:\Users\1pi\projects\watertap> idaes environment-info
Traceback (most recent call last):
  File "c:\users\1pi\miniconda3\envs\watertap-dev\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "c:\users\1pi\miniconda3\envs\watertap-dev\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\1pi\Miniconda3\envs\watertap-dev\Scripts\idaes.exe\__main__.py", line 4, in <module>
  File "c:\users\1pi\miniconda3\envs\watertap-dev\lib\site-packages\idaes\commands\__init__.py", line 18, in <module>
    from idaes.commands.base import command_base as cb
  File "c:\users\1pi\miniconda3\envs\watertap-dev\lib\site-packages\idaes\commands\base.py", line 79, in <module>
    def command_base(verbose, quiet):
  File "c:\users\1pi\miniconda3\envs\watertap-dev\lib\site-packages\click\decorators.py", line 304, in decorator
    return option(*(param_decls or ("--version",)), **attrs)(f)
  File "c:\users\1pi\miniconda3\envs\watertap-dev\lib\site-packages\click\decorators.py", line 192, in decorator
    _param_memo(f, OptionClass(param_decls, **option_attrs))
  File "c:\users\1pi\miniconda3\envs\watertap-dev\lib\site-packages\click\core.py", line 1714, in __init__
    Parameter.__init__(self, param_decls, type=type, **attrs)
TypeError: __init__() got an unexpected keyword argument 'package_name'

**Comment by lbianchi-lbl** [comment]
@aladshaw3 thanks. From a quick look at the `pip list` from your environment, I haven't seen anything that jumps to the eye, although some of the packages are slightly older if compared with e.g. the fresh environments from the latest CI job runs: 
![image](https://user-images.githubusercontent.com/48035537/155188102-892de030-2f0d-43cc-87e3-f5f1d9c10151.png)

One thing that would not be detected by this check is the exact version of IDAES and/or Pyomo that's installed, since there have been potentially relevant changes that have been merged in since you've created the environment: e.g. IDAES/idaes-pse#683, which most likely is not related to the test failures, but that particular change has not made its way through to the WaterTAP version of IDAES used in your environment, otherwise you would have Click 8 instead of 7 (which is also the cause of the `package_name` error; IDAES/idaes-pse#720 should close this temporary loophole by specifying Click 8 as a requirement).

**TL;DR** Unless someone else has more specific ideas on what could be causing these test failures, my suggestion would be to re-create the Conda environment from scratch (or create a new one in parallel) using the current WaterTAP `main`, and see if the issues persist. I do realize this is frustrating and time-consuming, unfortunately our dependency chain (WaterTAP <- IDAES <- Pyomo), which is already challenging to manage in ordinary situations, becomes even messier when a release is near.

**Comment by aladshaw3** [comment]
@lbianchi-lbl 

This fixed the issue. I had to create a new conda environment and install into that blank environment. 

So simply reinstalling in my existing environment with 
`pip uninstall watertap` 
`pip --no-cache-dir install -r requirements-dev.txt` 
was not enough. Should we add any language to the install instructions about this?

**Comment by bknueven** [comment]
@aladshaw3 I don't think adding language to the install instructions is necessary. I'm guessing your version of IDAES drifted slightly out of date.

Most users won't encounter issues like this because their environment isn't in a state of constant flux like a development environment is -- especially when we're running on pre-release versions of dependencies (like IDAES). I personally manage IDAES and Pyomo on my system by keeping them *also* set up in development mode (i.e., with cloned locally and installed with `pip install -e .` or `python setup.py develop`). This ensures I can keep track of the exact commit hash of IDAES and Pyomo I'm using.
